pdns/misc.hh

```diff
@@ -656,8 +657,20 @@ auto checked_conv(F from) -> T
   static_assert((std::numeric_limits<F>::is_signed && std::numeric_limits<T>::is_signed) || (!std::numeric_limits<F>::is_signed && !std::numeric_limits<T>::is_signed),
                 "checked_conv: The `T` and `F` types must either both be signed or unsigned");
 
-  if (from < std::numeric_limits<T>::min() || from > std::numeric_limits<T>::max()) {
-    throw std::out_of_range("checked_conv: conversion from value that is out of range for target type");
+  constexpr auto tMin = std::numeric_limits<T>::min();
+  if constexpr (std::numeric_limits<F>::min() != tMin) {
+    if (from < tMin) {
+      string s = boost::str(boost::format("checked_conv: source value %1% is smaller than target's minimum possible value %2%") % from % tMin);
```

Comment:
I would prefer not using `boost::format`here since we have had trouble with it calling `std:::ctype::narrow` which turned up not being thread-safe on some implementations, if I recall correctly.

Reply:
I switched the use to `std::to_string`. I force-pushed because it's small enough and can be merged immediately if you agree with it.